docker/apm-server/managed/main.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+func main() {
+	setup()
+	// create a reverse proxy to the APM Server running via Elastic Agent,
+	// headers are not rewritten, as not considered important right now.
+	target, _ := url.Parse("http://elastic-agent:8200")
+	http.Handle("/", httputil.NewSingleHostReverseProxy(target))
+	if err := http.ListenAndServe(":8200", nil); err != nil {
+		log.Fatal(err)
+		os.Exit(-1)
+	}
+}
+
+func setup() {
+	client := newKibanaClient()
+	if err := cleanup(client); err != nil {
+		log.Fatal(err)
+		os.Exit(-1)
+	}
+	if err := addAPMPolicy(client); err != nil {
+		log.Fatal(err)
+		os.Exit(-1)
+	}
+}
+
+type kibanaClient struct {
+	fleetURL string
+	pkgURL   string
+}
+type item struct {
+	ID string `json:"id"`
+}
+type items struct {
+	Items []item `json:"items"`
+}
+
+func newKibanaClient() *kibanaClient {
+	host := os.Getenv("KIBANA_HOST")
+	if host == "" {
+		host = "http://admin:changeme@localhost:5601"
+	}
+	return &kibanaClient{
+		fleetURL: fmt.Sprintf("%s/api/fleet", host),
+		pkgURL:   "https://epr-snapshot.elastic.co",
+	}
+}
+
+func cleanup(client *kibanaClient) error {
+	result, err := client.getPackagePolicies(fmt.Sprintf("kuery=ingest-package-policies.package.name:%s", "apm"))
+	if err != nil {
+		return errors.Wrap(err, "cleanup")
+	}
+	var ids []string
+	for _, item := range result.Items {
+		ids = append(ids, item.ID)
+	}
+	return client.deletePackagePolicies(ids)
+}
+
+func addAPMPolicy(client *kibanaClient) error {
+	agentPolicies, err := client.getAgentPolicies("kuery=ingest-agent-policies.is_default:true")
+	if err != nil {
+		return errors.Wrap(err, "addAPMPolicy")
+	}
+	if len(agentPolicies.Items) == 0 {
```

Comment:
Isn't there always a default policy if Elastic Agent is up?  I think this should keep retrying if the default policy is not found.

Reply:
also, what happens on container restart?  is there a way to check if we're already attached the default policy and do nothing in that case?

Reply:
added some retry logic - I can't seem to slow down the agent enough to run into this; please let me know if you still have issues.